File: 2021/10/24/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def countNodes(self, root: Optional[TreeNode]) -> int:
        self.r = 0
        def dfs(root):
            if root == None:
                return 0
            self.r += 1
            dfs(root.left)
            dfs(root.right)
        
        dfs(root)
        return self.r
            

File: 2021/10/24/main.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func countNodes(root *TreeNode) int {
	r := 0
	dfs(root, &r)
	return r
}

func dfs(root *TreeNode, r *int) {
	if root == nil {
		return
	}
	*r++
	dfs(root.Left, r)
	dfs(root.Right, r)
}
